### PR TITLE
(feat) add onIndexChange

### PR DESCRIPTION
**Pull request by pyrexfm**
Event handler for index change. Allows parent components to react to wizard step changes.

### CODE REVIEW

**Comment by pyrexfm** [comment]
Tackles #84

File: src/types.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ export type WizardProps = {
    * ```
    */
   wrapper?: React.ReactElement;
+  onIndexChange?: (stepIndex: number) => void;
```
Comment (b93rn) [inline src/types.ts:21]: onStepChange would be clearer in my opinion.

File: src/wizard.tsx [hunk 2]
```diff
@@ -16,6 +23,12 @@ const Wizard: React.FC<React.PropsWithChildren<WizardProps>> = React.memo(
     hasNextStep.current = activeStep < stepCount - 1;
     hasPreviousStep.current = activeStep > 0;
 
+    React.useEffect(() => {
```
Comment (b93rn) [inline src/wizard.tsx:26]: Not what useEffect is made for. Rather create a handleStepChangeFunction which includes the setActiveStep and replace all setActiveStep calls with handleStepChangeFunction. Than add onStepChange/onIndexChange tot that function.

File: src/wizard.tsx [hunk 3]
```diff
@@ -5,7 +5,14 @@ import { Handler, WizardProps } from './types';
 import WizardContext from './wizardContext';
 
 const Wizard: React.FC<React.PropsWithChildren<WizardProps>> = React.memo(
-  ({ header, footer, children, wrapper: Wrapper, startIndex = 0 }) => {
+  ({
+    header,
+    footer,
+    children,
+    onIndexChange,
```
Comment (b93rn) [inline src/wizard.tsx:12]: Callbacks below properties.

**Review by b93rn — CHANGES_REQUESTED** [review]

File: src/wizard.tsx (hunk 2 above)
Comment (devrnt) [reply src/wizard.tsx:26]: Indeed. We already keep track of the active index in the wizard. So let's extend the `goToPreviousStep` and `goToNextStep` methods and invoke the `onStepChange` callback

**Review by devrnt — CHANGES_REQUESTED** [review]

**Review by devrnt — APPROVED** [review]

**Comment by devrnt** [comment]
Thanks for this PR, I've added the final touches

**Pull request merged**